scio-smb/src/test/scala/com/spotify/scio/smb/SortMergeBucketTest.scala

```diff
@@ -0,0 +1,182 @@
+ */
+
+package com.spotify.scio.smb
+
+import com.spotify.scio.avro.{Account, Address, User}
+import com.spotify.scio.testing.PipelineSpec
+import com.spotify.scio.{Args, ContextAndArgs}
+import org.apache.beam.sdk.extensions.smb.AvroSortedBucketIO
+import org.apache.beam.sdk.values.{KV, TupleTag}
+
+import java.util.Collections
+import scala.jdk.CollectionConverters._
+
+trait SmbJob {
+
+  val keyClass: Class[Integer] = classOf
+  val keyField: String = "id"
+
+  def avroUsers(args: Args): AvroSortedBucketIO.Read[User] = AvroSortedBucketIO
+    .read(new TupleTag[User]("user"), classOf[User])
+    .from(args("users"))
+
+  def avroAccounts(args: Args): AvroSortedBucketIO.Read[Account] = AvroSortedBucketIO
+    .read(new TupleTag[Account]("account"), classOf[Account])
+    .from(args("accounts"))
+
+  def avroOutput(args: Args): AvroSortedBucketIO.Write[Integer, Void, User] = AvroSortedBucketIO
+    .write(keyClass, keyField, classOf[User])
+    .to(args("output"))
+
+  def avroTransformOutput(args: Args): AvroSortedBucketIO.TransformOutput[Integer, Void, User] =
+    AvroSortedBucketIO
+      .transformOutput(keyClass, keyField, classOf[User])
+      .to(args("output"))
+
+  def setUserAccounts(users: Iterable[User], accounts: Iterable[Account]): User = {
+    val u :: Nil = users.toList
+    setUserAccounts(u, accounts)
+  }
+
+  def setUserAccounts(user: User, accounts: Iterable[Account]): User = {
+    val sortedAccounts = accounts.toList
+      .sortBy(_.getAmount)(Ordering[java.lang.Double].reverse)
+      .asJava
+    User
+      .newBuilder(user)
+      .setAccounts(sortedAccounts)
+      .build()
+  }
+
+}
+
+object SmbJoinSaveJob extends SmbJob {
+
+  def main(cmdlineArgs: Array[String]): Unit = {
+    val (sc, args) = ContextAndArgs(cmdlineArgs)
+
+    sc.sortMergeJoin(
+      keyClass,
+      avroUsers(args),
+      avroAccounts(args)
+    ).values
+      .map { case (u, a) => setUserAccounts(u, Iterable(a)) }
+      .saveAsSortedBucket(avroOutput(args))
+
+    sc.run().waitUntilDone()
+  }
+
+}
+
+object SmbCoGroupSavePreKeyedJob extends SmbJob {
+
+  def main(cmdlineArgs: Array[String]): Unit = {
+    val (sc, args) = ContextAndArgs(cmdlineArgs)
+    sc.sortMergeCoGroup(
+      keyClass,
+      avroUsers(args),
+      avroAccounts(args)
+    ).map { case (id, (us, as)) => KV.of(id, setUserAccounts(us, as)) }
+      .saveAsPreKeyedSortedBucket(avroOutput(args))
+
+    sc.run().waitUntilDone()
+  }
+
+}
+
+object SmbTransformJob extends SmbJob {
+
+  def main(cmdlineArgs: Array[String]): Unit = {
+    val (sc, args) = ContextAndArgs(cmdlineArgs)
+
+    sc.sortMergeTransform(
+      keyClass,
+      avroUsers(args),
+      avroAccounts(args)
+    ).to(avroTransformOutput(args))
+      .via { case (_, (users, accounts), outputCollector) =>
+        outputCollector.accept(setUserAccounts(users, accounts))
+      }
+
+    sc.run().waitUntilDone()
+  }
+
+}
+
+class SortMergeBucketTest extends PipelineSpec {
```

Comment:
Small nit: maybe rename to `SortedBucketIOTest` to match other `ScioIO` test classes?